utils/zerovec/src/zerovec/slice.rs

```diff
@@ -0,0 +1,318 @@
+        // This is safe because ZeroSlice is transparent over [T::ULE]
+        // so Box<ZeroSlice<T>> can be safely cast from Box<[T::ULE]>
+        unsafe { Box::from_raw(Box::into_raw(slice) as *mut Self) }
+    }
+
+    /// Returns this slice as its underlying `&[u8]` byte buffer representation.
+    ///
+    /// Useful for serialization.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    ///
+    /// // The little-endian bytes correspond to the numbers on the following line.
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let nums: &[u16] = &[211, 281, 421, 461];
+    ///
+    /// let zerovec = ZeroVec::alloc_from_slice(nums);
+    ///
+    /// assert_eq!(bytes, zerovec.as_bytes());
+    /// ```
+    #[inline]
+    pub fn as_bytes(&self) -> &[u8] {
+        T::ULE::as_byte_slice(self.as_slice())
+    }
+
+    /// Dereferences this slice as `&[T::ULE]`.
+    #[inline]
+    pub fn as_slice(&self) -> &[T::ULE] {
+        &self.0
+    }
+
+    /// Returns the number of elements in this slice.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    /// use zerovec::ule::AsULE;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    ///
+    /// assert_eq!(4, zerovec.len());
+    /// assert_eq!(
+    ///     bytes.len(),
+    ///     zerovec.len() * std::mem::size_of::<<u16 as AsULE>::ULE>()
+    /// );
+    /// ```
+    #[inline]
+    pub fn len(&self) -> usize {
+        self.as_slice().len()
+    }
+
+    /// Returns whether this slice is empty.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    /// assert!(!zerovec.is_empty());
+    ///
+    /// let emptyvec: ZeroVec<u16> = ZeroVec::parse_byte_slice(&[]).expect("infallible");
+    /// assert!(emptyvec.is_empty());
+    /// ```
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.as_slice().is_empty()
+    }
+}
+
+impl<T> ZeroSlice<T>
+where
+    T: AsULE,
+{
+    /// Gets the element at the specified index. Returns None if out of range.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    ///
+    /// assert_eq!(zerovec.get(2), Some(421));
+    /// assert_eq!(zerovec.get(4), None);
+    /// ```
+    #[inline]
+    pub fn get(&self, index: usize) -> Option<T> {
+        self.as_slice().get(index).copied().map(T::from_unaligned)
+    }
+
+    pub(crate) fn get_ule_ref(&self, index: usize) -> Option<&T::ULE> {
```

Comment:
Suggestion: Elango found a use case for this function in his Script_Extensions work.  Please make it public.